src/compiler/types.ts

```diff
@@ -5825,19 +5825,20 @@ namespace ts {
 
     export const enum InferencePriority {
         NakedTypeVariable            = 1 << 0,  // Naked type variable in union or intersection type
-        SpeculativeTuple             = 1 << 1,  // Speculative tuple inference
-        SubstituteSource             = 1 << 2,  // Source of inference originated within a substitution type's substitute
-        HomomorphicMappedType        = 1 << 3,  // Reverse inference for homomorphic mapped type
-        PartialHomomorphicMappedType = 1 << 4,  // Partial reverse inference for homomorphic mapped type
-        MappedTypeConstraint         = 1 << 5,  // Reverse inference for mapped type
-        ContravariantConditional     = 1 << 6,  // Conditional type in contravariant position
-        ReturnType                   = 1 << 7,  // Inference made from return type of generic function
-        LiteralKeyof                 = 1 << 8,  // Inference made from a string literal to a keyof T
-        NoConstraints                = 1 << 9,  // Don't infer from constraints of instantiable types
-        AlwaysStrict                 = 1 << 10,  // Always use strict rules for contravariant inferences
-        MaxValue                     = 1 << 11, // Seed for inference priority tracking
-
-        PriorityImpliesCombination = ReturnType | MappedTypeConstraint | LiteralKeyof,  // These priorities imply that the resulting type should be a combination of all candidates
+        TemplateLiteralPlaceholder   = 1 << 1,  // Inference to a template literal type placeholder

```

Comment:
@ahejlsberg I'm not 100% confident in the priority here relative to other priorities. Please advise if you feel this needs to change.

Reply:
Personally, I usually just inject new priorities above `ReturnType` - anywhere up there is usually fine, since `ReturnType` is usually the only one it really _has_ to beat (since we regularly compare return type inference results with parameter inference results). These inferences are probably better than most of the other reduced priority locations, so putting it at the top seems pretty sensible.